### PR TITLE
Refactor attributes in PlanNode

### PR DESCRIPTION
**Pull request by liuminghui233**
Refactor attributes in PlanNode.

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/FilterNullNodeSerdeTest.java [hunk 1]
```diff
@@ -104,7 +101,10 @@ public void TestSerializeAndDeserialize() throws IllegalPathException {
 
     FilterNullNode filterNullNode =
         new FilterNullNode(
-            new PlanNodeId("TestFilterNullNode"), filterNode, FilterNullPolicy.ALL_NULL, null);
+            new PlanNodeId("TestFilterNullNode"),
+            filterNode,
+            FilterNullPolicy.ALL_NULL,
+            ImmutableList.of());
```
Comment (JackieTien97) [inline server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/FilterNullNodeSerdeTest.java:107]: Should not be `ImmutableList.of()`

File: server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/GroupByLevelNodeSerdeTest.java [hunk 2]
```diff
@@ -106,7 +103,10 @@ public void TestSerializeAndDeserialize() throws IllegalPathException {
 
     FilterNullNode filterNullNode =
         new FilterNullNode(
-            new PlanNodeId("TestFilterNullNode"), filterNode, FilterNullPolicy.ALL_NULL, null);
+            new PlanNodeId("TestFilterNullNode"),
+            filterNode,
+            FilterNullPolicy.ALL_NULL,
+            ImmutableList.of());
```
Comment (JackieTien97) [inline server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/GroupByLevelNodeSerdeTest.java:109]: same as above

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/source/SeriesAggregateScanNode.java [hunk 3]
```diff
@@ -67,28 +67,31 @@
...
  */
-public class SeriesAggregateScanNode extends SourceNode implements IOutputPlanNode {
+public class SeriesAggregateScanNode extends SourceNode {
 
   // The series path and aggregation functions on this series.
   // (Currently, we only support one series in the aggregation function)
   private final PartialPath seriesPath;
-  private final List<AggregationType> aggregateFuncList;
+  @Deprecated private final List<AggregationType> aggregateFuncList;
+  private Aggregation aggregation;
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/source/SeriesAggregateScanNode.java:76]: fixed

File: server/src/main/java/org/apache/iotdb/db/mpp/common/FilterNullParameter.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  public List<InputLocation> getFilterNullColumns() {
+    return filterNullColumns;
+  }
+
+  public void serialize(ByteBuffer byteBuffer) {
+    ReadWriteIOUtils.write(filterNullPolicy.ordinal(), byteBuffer);
+    if (filterNullColumns == null) {
+      ReadWriteIOUtils.write(-1, byteBuffer);
+    } else {
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/common/FilterNullParameter.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/QueryPlanBuilder.java [hunk 5]
```diff
@@ -270,10 +272,8 @@ public void planFilterNull(FilterNullComponent filterNullComponent) {
         new FilterNullNode(
             context.getQueryId().genPlanNodeId(),
             this.getRoot(),
-            filterNullComponent.getWithoutPolicyType(),
-            filterNullComponent.getWithoutNullColumns().stream()
-                .map(Expression::getExpressionString)
-                .collect(Collectors.toList()));
+            new FilterNullParameter(
+                filterNullComponent.getWithoutPolicyType(), ImmutableList.of()));
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/QueryPlanBuilder.java:276]: fixed

File: server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/FilterNullNodeSerdeTest.java (hunk 1 above)
Comment (liuminghui233) [reply server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/FilterNullNodeSerdeTest.java:107]: fixed

File: server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/GroupByLevelNodeSerdeTest.java (hunk 2 above)
Comment (liuminghui233) [reply server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/node/process/GroupByLevelNodeSerdeTest.java:109]: fixed

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**